lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -458,6 +460,135 @@ def _email_opt_in_checkbox(self, response, org_name_string=None):
+        url = reverse('submit_financial_assistance_request')
+        return self.client.post(url, json.dumps(data), content_type='application/json')
+
+    @patch.object(views, '_record_feedback_in_zendesk')
+    def test_submit_financial_assistance_request(self, mock_record_feedback):
+        username = self.user.username
+        course = unicode(self.course_key)
+        legal_name = 'Jesse Pinkman'
+        country = 'United States'
+        income = '1234567890'
+        reason_for_applying = "It's just basic chemistry, yo."
+        goals = "I don't know if it even matters, but... work with my hands, I guess."
+        effort = "I'm done, okay? You just give me my money, and you and I, we're done."
+        data = {
+            'username': username,
+            'course_id': course,
+            'legal_name': legal_name,
+            'email': self.user.email,
+            'country': country,
+            'income': income,
+            'reason_for_applying': reason_for_applying,
+            'goals': goals,
+            'effort': effort,
+            'marketing_permission': False,
+        }
+        response = self._submit_financial_assistance_form(data)
+        self.assertEqual(response.status_code, 204)
+
+        __, ___, ticket_subject, ticket_body, tags, additional_info = mock_record_feedback.call_args[0]
+        for info in (country, income, reason_for_applying, goals, effort):
+            self.assertIn(info, ticket_body)
+        self.assertIn('This user HAS NOT allowed this content to be used for edX marketing purposes.', ticket_body)
+
+        self.assertEqual(
+            ticket_subject,
+            'Financial assistance request for user {username} in course {course}'.format(
+                username=username,
+                course=course
+            )
+        )
+        self.assertDictContainsSubset(
+            {
+                'issue_type': 'Financial Assistance',
+                'course_id': course
+            },
+            tags
+        )
+        self.assertIn('Client IP', additional_info)
+
+    @patch.object(views, '_record_feedback_in_zendesk', return_value=False)
+    def test_zendesk_submission_failed(self, _mock_record_feedback):
+        response = self._submit_financial_assistance_form({
+            'username': self.user.username,
+            'course_id': '',
+            'legal_name': '',
+            'email': '',
+            'country': '',
+            'income': '',
+            'reason_for_applying': '',
+            'goals': '',
+            'effort': '',
+            'marketing_permission': False,
+        })
+        self.assertEqual(response.status_code, 500)
+
+    def test_submit_financial_assistance_missing_fields(self):
+        response = self._submit_financial_assistance_form({})
+        self.assertEqual(response.status_code, 400)
+
+    def test_submit_financial_assistance_wrong_username(self):
+        response = self._submit_financial_assistance_form({'username': 'wwhite'})
```

Comment:
Might be good to use ddt here for this method and the one above. Not sure if we would also want to add other cases using ddt